packages/carousel/src/react/index.tsx

```diff
@@ -221,25 +221,41 @@ interface PageProps extends HTMLPropsFor<'ul'> {
   paged?: boolean
   isActivePage?: boolean
 }
-const Page: React.FC<PageProps> = props => {
-  const ref = React.useRef<HTMLUListElement>(null)
 
+const Page: React.FC<PageProps> = props => {
   const { children, isActivePage, paged, ...rest } = props
   const { offset, transitioning, setTransitioning } = React.useContext(
     CarouselContext
   )
-  React.useEffect(() => {
-    if (paged && isActivePage) {
-      const page = ref.current
-      if (page) {
-        const focusFirstChild = (evt: TransitionEvent) => {
-          setTransitioning(false)
-          evt.target === page && (page.firstElementChild as HTMLElement).focus()
+
+  const ref = React.useRef<HTMLUListElement>(null)
+
+  React.useEffect(
+    function responseToTransitionEnd() {
+      const { current: page } = ref
+      if (!page || !paged || !isActivePage) return
+
+      const onTransitionEnd = (evt: TransitionEvent) => {
+        const isPageTransition =
+          evt.target === page && evt.propertyName === 'transform'
```

Comment:
The issue was caused because we were responding to all self and child transitions. The bug in the example was caused because we were updating state when a transition in the Card(a opacity transition in Card.Actions) was setting the new transition state prematurely and hiding the previous page.

The fix is to only update the transition state if the `transitionend` matches the current page as the event target and only on `transform` transitions

Reply:
wow, nice find!